### PR TITLE
MSA Library

### PR DESCRIPTION
**Pull request by joergi-w**
This PR implements a data structure for the T-Coffee library and provides tests and documentation.

### CODE REVIEW

File: include/seqan3/alignment/multiple/library.hpp [hunk 1]
```diff
@@ -0,0 +1,427 @@
...
+            return iter->second.try_emplace(pos, score).second;
+    }
+
+    /*!\brief Add a score to a possibly existing entry in the library.
+     * \param seq A pair of sequence indeices.
+     * \param pos A pair of position indices.
+     * \param score The score that is added to the specified position in the specified sequence pair.
+     *              If this location does not exist it will be created with value `score`.
+     */
+    void add(coord_type seq, coord_type pos, score_type score)
```
Comment (smehringer) [inline include/seqan3/alignment/multiple/library.hpp]: question: isn't `add()` equivalent to `operator[{seq1, seq2, pos1, po2}] += score` ?

File: include/seqan3/alignment/multiple/library.hpp [hunk 2]
```diff
@@ -0,0 +1,427 @@
...
+        if (!inserted)   // append to an existing sequence pair
+            iter->second[pos] += score;
+    }
+
+    /*!
+     * \brief Retrieve the iterator that contains the score of pairing two positions in the specified sequences.
+     * \param seq_pos A quadruple consisting of (seq1, seq2, pos1, pos2).
+     * \return An iterator to the score if the specified entry exists and end() otherwise.
+     */
+    iterator operator[](std::tuple<size_t, size_t, size_t, size_t> seq_pos) noexcept
```
Comment (smehringer) [inline include/seqan3/alignment/multiple/library.hpp]: I think an  `operator[]` should always return a reference to a value (not an iterator)?

File: include/seqan3/alignment/multiple/library.hpp [hunk 3]
```diff
@@ -0,0 +1,427 @@
...
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    constexpr iterator begin() noexcept
+    {
+        return iterator(this);
```
Comment (smehringer) [inline include/seqan3/alignment/multiple/library.hpp]:
```suggestion
        return iterator{this};
```
since this is a construction right?

**Comment by rrahn** [comment]
Hi @joergi-w, do you think you can work on this until the end of next week, such that this is in a good state for merging, or do you think it needs more time? If latter is the case, I'd like to ask you to close this PR until it is back in your scope.

**Comment by joergi-w** [comment]
I'm closing it, because I have started from scratch in another branch after we discussed the graph-based design and it does not make sense to base on this code. I'm going to create a new PR with the graph.

**Comment by rrahn** [comment]
Great, but please do not open the new PR in the next week. We are trying to get all open stuff done as soon as possible and then move on to a little different organisation structure. I write more to it on Gitter.

**Pull request closed without merging**